components/script/dom/bindings/trace.rs

```diff
@@ -160,6 +160,50 @@ impl<T: JSTraceable> JSTraceable for Rc<T> {
     }
 }
 
+impl<T: JSTraceable> JSTraceable for Arc<T> {
+    fn trace(&self, trc: *mut JSTracer) {
+        (**self).trace(trc)
+    }
+}
+
+impl JSTraceable for Mutex<Option<SharedRt>> {
```

Comment:
We should be able to implement it on concrete types, just not generic ones (`impl<T> JSTraceable for Foo<T>`). When we add the lint check it can recurse type parameters of the concrete types.